### PR TITLE
Android fullscreen fix for v1 branch

### PR DESCRIPTION
**Pull request by lucianomlima**

### CODE REVIEW

**Comment by davidohayon669** [comment]
will take a look at it soon
thanks

**Comment by hasaezs** [comment]
@davidohayon669 You could check this? It is necessary to fix that.
Cheers.

**Comment by davidohayon669** [comment]
@lucianomlima I took a look at this issue and I'm not positive you solved it correctly.

A few things:
- Back button action (At least with React-Native as far as I know, I have little experience with native Android) will close the app unless any other actions (mainly routing actions) were registered to it with an event lister (With ReactNative.BackAndroid). On the example app, hitting back from a fullscreen will close the app. On the app I work on, hitting back will perform the registered navigation action (which in my case is a 'pop' of the current scene) and the YouTube player will shut down because its unmounted.
- When I disable the default back action in the example app, by adding `BackAndroid.addEventListener('hardwareBackPress', () => true);`, nothing will happen when hitting back.

Are you sure your use case only closes the fullscreen mode or is it also performing a navigation event?

As I understand, to have the back button **only** exit the fullscreen mode, `onBackPressed` should be overridden on YouTubeView.

I will further take a look at it when I improve my app but I can't merge this request right now as it only solve part of the problem and leaves the Back button issue still buggy

**Comment by lucianomlima** [comment]
Thanks @davidohayon669.
In my tests, I was able to identify the following behaviors:

I open my app, navigate to a videos list (thumbnails), tap a video, open the player in fullscreen.

1) If I rotate the cell phone to landscape and then return to portrait, and my device have automatic rotation enabled, the orientation returns to portrait.

2) If the device has automatic rotation turned off, the application remains in landscape.

3) If the application is in fixed portrait orientation in AndroidManifest.xml, the application remains in the landscape.

4) If instead of rotating the device, I press the device's back button, it exits the fullscreen but keeps the orientation locked in the landscape, regardless of the application orientation configuration and the automatic rotation of the device.

With this, I has concluded that the problem was not the orientation because in case one, without pressing the back button, the YouTubeView was unmounted and the automatic rotation worked correctly. Only when you go back through the device's  back button that blocks the orientation.

I confess I did not try to exit the fullscreen with an override of the onBackPressed method, but I try this and I will comment later.

I did not quite understand the case of the default back action disabled. What would be the behavior you expected?

**Comment by davidohayon669** [comment]
I'm afraid that what you interpret as an "exit fullscreen" action triggered by the back button is a side effect of an unrelated action that is registered with the back button, probably one that destroys the view where the player is mounted so it destroys it as well.

The android YouTube API doesn't mention anything about a default back button action, It only says that any exiting of a fullscreen mode will return the orientation back to the intended state (as happens after hitting the fullscreen icon while on fullscreen)

I would try to implement something like
`@Override
public void onBackPressed() {
    if (mYoutubeController.isPlayerFullscreen()) {
        mYoutubeController.setFullscreen(false);
    } else {
        super.onBackPressed();
    }
}`
in YouTubeView.java

But for this we also need to implement a `YouTubePlayer.OnFullscreenListener` on the player so we can update the controller with the fullscreen state

tl;dr
The back button doesn't naturally exits fullscreen. It does what it is registered to do at the same moment and the exiting of the player happens as a result of that
